extensions/common/src/main/java/com/twocoders/extensions/common/LogExtensions.kt

```diff
@@ -0,0 +1,46 @@
+package com.twocoders.extensions.common
+
+import android.content.Context
+import android.util.Log
+import java.io.BufferedWriter
+import java.io.File
+import java.io.FileWriter
+import java.io.IOException
+import java.text.SimpleDateFormat
+import java.util.*
+
+private const val PATTERN_LOG_TO_FILE = "yyyy-MM-dd HH:mm:ss.SSS"
+
+fun Any.logd(message: String) {
+    Log.d(this::class.java.simpleName, message)
+}
+
+fun Any.logd(message: String, tr: Throwable) {
+    Log.d(this::class.java.simpleName, message, tr)
+}
+
+fun Any.logToFile(context: Context, message: String) {
```

Comment:
vacsinou pouivas iba `message` a `this` maximalne na tag. Nebolo by lepsie do textu zahrnut aj `this.toString()`? Potom by si vedel ake hodnoty su napriklad v tom objekte a bolo by to asi uzitocnejsie.

Je to len navrh, neviem presne aky ucel maju mat tieto log metody, len mi pride, ze minimalne tato `toFile`, kedze nevyuziva vobec `this`, tak ani nemusi byt extension. Staci ze bude global a hocikde ju mozes volat aj bez instancie

Reply:
ty metody měli jen usnadnit logování tím způsobem, že nemusíš stále doplňovat název objektu/třídy. Vytáhne si to za tebe. A `logToFile` používá `this` v těch catch blocích na zalogování errorů. Nevím, zda bych tam teď ten `this.toString()` dával, můžeme přidat v další verzi. U větších objektů s komplikovanějšími membery bez overridnutých `toString` by pak ten log nemusel vypadat úplně přívětivě. Zkusme se na toto zaměřit do budoucna jako improvement a trochu to ještě promyslet.